src/Shared/Traits.cs

```diff
@@ -77,6 +77,11 @@ public Traits()
         /// </summary>
         public readonly bool LogPropertyFunctionsRequiringReflection = !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("MSBuildLogPropertyFunctionsRequiringReflection"));
 
+        /// <summary>
+        /// Track the environment variables actually read and used.
+        /// </summary>
+        public readonly bool TrackReadsOnEnvironmentVariables = !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("MSBUILDTRACKENVVARREADS"));
```

Comment:
It's nice to compare against 1 rather than null/empty because that makes it possible to add other values in future without breaking existing users.

Reply:
Changed in latest commit.